hazelcast/include/hazelcast/client/config/SocketOptions.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef HAZELCAST_CLIENT_CONFIG_SOCKETOPTIONS_H_
+#define HAZELCAST_CLIENT_CONFIG_SOCKETOPTIONS_H_
+
+#include "hazelcast/util/HazelcastDll.h"
+
+#if  defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#pragma warning(push)
+#pragma warning(disable: 4251) //for dll export	
+#endif
+
+namespace hazelcast {
+    namespace client {
+        namespace config {
+            /**
+             * TCP Socket options
+             */
+            class HAZELCAST_API SocketOptions {
+            public:
+                /**
+                 * constant for kilobyte
+                 */
+                static const int KILO_BYTE = 1024;
+
+                /**
+                 * default buffer size of Bytes
```

Comment:
also this one

Reply:
from java client again. https://github.com/hazelcast/hazelcast/blob/master/hazelcast-client/src/main/java/com/hazelcast/client/config/SocketOptions.java#L30